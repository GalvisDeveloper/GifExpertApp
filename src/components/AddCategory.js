

import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(catg => [inputValue, ...catg, ]);
            setInputValue("");
        } else {
            alert("Añada una categoria con nombre más largo", "Alerta");
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                placeholder="Ingrese una categoría para generar GIFs de ésta"
            />
        </form>
    );

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;