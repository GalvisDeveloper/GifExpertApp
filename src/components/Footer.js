

import React from 'react'
import PropTypes from 'prop-types';
import GitHubIcon from '@material-ui/icons/GitHub';

export const Footer = ({ property }) => {
    return (
        <div className="footer">
            <h1 className="animate__animated animate__bounce animate__zoomInUp">{property + " Cristian Bustos ➡ "}</h1>
            <GitHubIcon />
            <a href="https://github.com/GalvisDeveloper?tab=repositories"> GalvisDeveloper </a>
            <h1> - 2020 © </h1>
            <br />
        </div>
    )
}

Footer.prototype = {
    property: PropTypes.string.isRequired
}

export default Footer;