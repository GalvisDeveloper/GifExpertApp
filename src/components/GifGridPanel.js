

import React from 'react'
import PropTypes from 'prop-types';

export const GifGridPanel = ({ url, title }) => {

    return (
        <div className="card animate__animated callout-subtitle animate__animated animate__zoomInDown">
            <p>Gif: {title}</p>
            <img src={url} alt={title} />
        </div>
    )
}

GifGridPanel.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
