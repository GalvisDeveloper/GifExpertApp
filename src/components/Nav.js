

import React from 'react'
import PropTypes from 'prop-types';

const Nav = ({ message }) => {
    return (
        <div className="nav">
            <h1 className="animate__animated animate__bounce animate__zoomInUp">{message}</h1>
        </div>
    )
}

Nav.prototype = {
    message: PropTypes.string.isRequired
}

export default Nav;