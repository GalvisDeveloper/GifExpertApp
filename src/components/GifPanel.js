

import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridPanel } from './GifGridPanel';
import PropTypes from 'prop-types';

const GifPanel = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__bounce animate__rotateIn">{category}</h3>
            { loading && <p className="animate__animated animate__bounce animate__flash">Loading...</p>}
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridPanel
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifPanel.propTypes = {
    category: PropTypes.string.isRequired,
}

export default GifPanel;