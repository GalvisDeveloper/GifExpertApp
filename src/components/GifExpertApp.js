

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import AddCategory from './AddCategory';
import { GifPanel } from './GifPanel';

const GifExpertApp = ({ title }) => {

    const [categories, setCategories] = useState(['Motogp']);

    return (
        <div className="container">
            <div className="title">
                <h2 className="animate__animated animate__bounce animate__slideInDown animate_fast">{title}</h2>
            </div>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map(category => (
                        <GifPanel
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </div>
    );
}

GifExpertApp.prototype = {
    title: PropTypes.string.isRequired
}

export default GifExpertApp;
