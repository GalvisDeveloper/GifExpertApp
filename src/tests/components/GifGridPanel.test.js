import React from 'react';
const { shallow } = require("enzyme")
import '@testing-library/jest-dom';
import { GifGridPanel } from '../../components/GifGridPanel';


describe('Pruebas en el <GifGridPanel />', () => {

    const title = "Titulo prueba";
    const url = "https://localhost/algo.jpg";
    const wrapper = shallow(<GifGridPanel url={url} title={title} />);

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe contener un parrafo con el titulo y el (Gif:) antes de el', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe("Gif: " + title);
    })

    test('Testeando la imagen', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    })

    test('Debe tener (animate__zoomInDown)', () => {
        const div = wrapper.find('div');
        const classDiv = div.prop('className');

        expect(classDiv.includes('animate__zoomInDow')).not.toBe(false);
    })

})