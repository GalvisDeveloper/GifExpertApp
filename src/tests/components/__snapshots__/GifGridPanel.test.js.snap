// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGridPanel /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated callout-subtitle animate__animated animate__zoomInDown"
>
  <p>
    Gif: 
    Titulo prueba
  </p>
  <img
    alt="Titulo prueba"
    src="https://localhost/algo.jpg"
  />
</div>
`;
